app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarFragment.kt

```diff
@@ -514,6 +552,59 @@ class CalendarFragment :
             .show()
     }
 
+    private fun setMonthView() {
+        monthYearText.text = monthYearFromDate(selectedDate)
+        val daysInMonth: ArrayList<String> = daysInMonthArray(selectedDate)
+
+        val eventMap = calendarController.getEventsForMonth(selectedDate)
+
+        if (!::monthViewAdapter.isInitialized) {
+            monthViewAdapter = MonthViewAdapter(daysInMonth, eventMap)
+            monthRecyclerView.adapter = monthViewAdapter
+        } else {
+            monthViewAdapter.updateData(daysInMonth, eventMap)
+        }
+
+        val layoutManager: RecyclerView.LayoutManager = GridLayoutManager(requireContext(), 7)
+        monthRecyclerView.layoutManager = layoutManager
+    }
+
+    private fun daysInMonthArray(date: LocalDate): ArrayList<String> {
+        val daysInMonthArray: ArrayList<String> = ArrayList()
+        val yearMonth: YearMonth = YearMonth.of(date.year, date.monthOfYear)
+        val daysInMonth: Int = yearMonth.lengthOfMonth()
+        val firstOfMonth: LocalDate = date.withDayOfMonth(1)
+        var dayOfWeek: Int = firstOfMonth.dayOfWeek().get()
+        if (dayOfWeek == 7) {
```

Comment:
add a constant for what 7 represents

Reply:
I am sorry, this does not clear up my confusion: Why is `7` mapped to `0`?

Reply:
Sry, this part is not necessary. My fault.

Reply:
As in the whole if statement can be deleted?